src/components/robot/Robot.ts

```diff
@@ -0,0 +1,195 @@
+import type {
+  PoseInFrame,
+  ResourceName,
+  Transform,
+} from '../../gen/common/v1/common_pb.esm';
+import type { Duration } from 'google-protobuf/google/protobuf/duration_pb';
+import type { Extra } from '../../types';
+import type proto from '../../gen/robot/v1/robot_pb.esm';
+
+export interface Robot {
+  /**
+   * Get the list of operations currently running on the robot.
+   *
+   * @privateRemarks
+   *   TODO: this function should return an more idiomatic type instead of just
+   *   passing along a proto type.
+   * @group Operations
+   * @alpha
+   */
+  getOperations(): Promise<proto.Operation[]>;
```

Comment:
returning proto types isn't inherently a bad thing. The operation is a POD, so unless its API is really bad, this is O.K.

Reply:
what is "POD"?


i can see it both ways. my main concern is coupling the proto api to the sdk api, especially since we are interested in potentially moving to another protobuf generator at some point. but yeah, it's also annoying to define interfaces that are essentially pass-throughs.

not necessarily something we need to commit on yet, but should as we move to a stable release

Reply:
POD is a plain ol datatype.